tests/frameworks/layers/l2l4_network_creation_test.py

```diff
@@ -193,6 +193,52 @@ def testL4L2ColumnCreate(self):
     net.run(3)
 
 
+  def testL4L2ColumnLinks(self):
+    """
+    In this simplistic test we create a network and ensure that it has the
+    correct links between regions.
+    """
+
+    # Create a simple network to check its architecture
+    net = createNetwork(networkConfig1)
+
+    links = net.getLinks()
+
+    # Make sure that we have the right number before going on to specifics
+    self.assertEqual(len(list(net.getLinks())), 6, "Incorrect number of links")
+
+    # These are all the links we're hoping to find
+    desired_links=[("sensorInput_0", "dataOut", "L4Column_0", "activeColumns"),
+      ("L2Column_0", "feedForwardOutput", "L4Column_0", "apicalInput"),
+      ("externalInput_0", "dataOut", "L4Column_0", "basalInput"),
+      ("L4Column_0", "predictedActiveCells", "L2Column_0",
+       "feedforwardGrowthCandidates"),
+      ("L4Column_0", "activeCells", "L2Column_0", "feedforwardInput"),
+      ("sensorInput_0", "resetOut", "L2Column_0", "resetIn")]
+
+    # This gets the actual C-based link objects and discards their tags
+    links = map(lambda x: x.second, links)
+
+    # Form the links into a text-based representation
+    links = map(lambda x: (x.getSrcRegionName(),
+    x.getSrcOutputName(), x.getDestRegionName(), x.getDestInputName()), links)
+
+    # Build a descriptive error message to pass to the user
+    error_message = "Links incorrectly formed in simple L2L4 network: \n"
+    for link in desired_links:
+      if not link in links:
+        error_message += "Failed to find link: {}.{} to {}.{} \n".format(
+          link[0],link[1],link[2],link[3])
+
+    for link in links:
+      if not link in desired_links:
+        error_message += "Found unexpected link: {}.{} to {}.{} \n".format(
+          link[0],link[1],link[2],link[3])
+
+    self.assertEqual(sorted(desired_links), sorted(links),
```

Comment:
Better to us `assertSequenceEqual` when comparing two lists. And if you go the set approach, this becomes something like:

```
message = "Actual links:\n\t{}\ndon't match expected:\n\t{}".format(links, desired_links)"
self.assertSetEqual(desired_links, links, message)
```

If the message format doesn't look good then you can construct something nicer but I'm guessing this will be sufficient.